src/components/AttachmentPicker/index.native.js

```diff
@@ -94,43 +119,117 @@ function getDataForUpload(fileData) {
     };
 }
 
-const propTypes = {
-    children: PropTypes.func.isRequired,
-};
-
 /**
  * This component renders a function as a child and
  * returns a "show attachment picker" method that takes
- * a callback. This is the web/mWeb/desktop version since
- * on iOS Safari we must append a hidden input to the DOM
- * and listen to onChange event. When the show method is
- * called an attachment
- *
- * @example
- * <AttachmentPicker>
- * {({openPicker}) => (
- *     <Button
- *         onPress={() => {
- *             openPicker({
- *                 onPicked: (file) => {
- *                     // Display or upload File
- *                 },
- *             });
- *         }}
- *     />
- * )}
- * </AttachmentPicker>
- *
- * @returns {Function}
+ * a callback. This is the ios/android implementation
+ * opening a modal with attachment options
  */
-const AttachmentPicker = ({children}) => children({
-    openPicker: ({onPicked}) => {
-        show((response) => {
-            onPicked(getDataForUpload(response));
+class AttachmentPicker extends Component {
+    constructor(...args) {
+        super(...args);
+
+        this.state = {isVisible: false, onPicked: () => {}, result: null};
+
+        this.menuItemData = [
+            {
+                icon: Camera,
+                text: 'Take Photo',
+                pickAttachment: () => showImagePicker(RNImagePicker.launchCamera),
+            },
+            {
+                icon: Gallery,
+                text: 'Choose from Gallery',
+                pickAttachment: () => showImagePicker(RNImagePicker.launchImageLibrary),
+            },
+            {
+                icon: Paperclip,
+                text: 'Choose Document',
+                pickAttachment: showDocumentPicker,
+            },
+        ];
+
+        this.setResult = this.setResult.bind(this);
+        this.close = this.close.bind(this);
+        this.onModalHide = this.onModalHide.bind(this);
+    }
+
+    /**
+     * After the modal closes, if an attachment was selected delegate it to the `onPicked` callback
+     */
+    onModalHide() {
+        if (this.state.result) {
+            this.state.onPicked(this.state.result);
+        }
+    }
+
+    /**
+     * Store the selected attachment mapped to an appropriate file interface
+     *
+     * @param {ImagePickerResponse|DocumentPickerResponse} result
+     */
+    setResult(result) {
+        if (result && !result.didCancel && !result.error) {
+            this.setState({result: getDataForUpload(result)});
+        }
+    }
+
+    open(onPicked) {
+        if (typeof onPicked !== 'function') {
+            throw new Error(
+                `Invalid onPicked parameter. Check the children passed to: ${AttachmentPicker.displayName}`,
+            );
+        }
+
+        this.setState({isVisible: true, onPicked, result: null});
+    }
+
+    close() {
+        this.setState({isVisible: false});
+    }
```

Comment:
Could you add function docs here please

Reply:
I pushed a fix, but am leaving the comment unresolved so you can verify this fix properly address your comment